src/ConfigTemplateDefinitionResolver.php

```diff
@@ -0,0 +1,435 @@
+        $config_template_merge = $config_template_type_plugin->getConfigChangesForField($this->collection->getId(), $entity_type, $bundle, $field_name, $source_field_config);
+        $config = $config_template_merge->applyMerge($config, $dynamic_field_definition['path']);
+      }
+
+      if (!empty($field_map['templates'])) {
+        $this->createAllFieldConfigurations($entity_type, $bundle, $field_definition, $field_map['templates']);
+      }
+    }
+
+    return $config;
+  }
+
+  /**
+   * Create field related configurations.
+   *
+   * @param string $entity_type
+   *   The entity type.
+   * @param string $bundle
+   *   The bundle name.
+   * @param \Drupal\Core\Field\FieldDefinitionInterface $field_definition
+   *   The field definition.
+   * @param array $field_templates
+   *   The list of configuration templates for field.
+   *
+   * @throws \Exception
+   */
+  protected function createAllFieldConfigurations(string $entity_type, string $bundle, FieldDefinitionInterface $field_definition, array $field_templates) {
+    $field_collection = ConfigTemplateDefinitionCollection::createFromTemplates($this->collection->getId(), $this->entityType, $field_templates, $this->collection->getDirectory());
+    $config_resolver = ConfigTemplateDefinitionResolver::create(\Drupal::getContainer(), $field_collection);
+
+    foreach ($field_collection->getDefinitionNames() as $definition_name) {
+      $field_configuration_template = $field_collection->getDefinition($definition_name);
+
+      $this->fieldTemplatesQueue[] = [
+        'config_resolver' => $config_resolver,
+        'field_configuration_template' => $field_configuration_template,
+        'bundle' => $bundle,
+        'field_name' => $field_definition->getName(),
+      ];
+    }
+  }
+
+  /**
+   * Create configuration for field.
+   *
+   * @param \Drupal\testsite_builder\ConfigTemplateDefinition $configuration_template
+   *   The configuration template array.
+   * @param string $entity_type
+   *   The entity type.
+   * @param string $bundle
+   *   The bundle.
+   * @param string $field_name
+   *   The field name.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public function createConfigForField(ConfigTemplateDefinition $configuration_template, string $entity_type, string $bundle, string $field_name) {
+    $this->templateDefinition = $configuration_template;
+
+    $dynamic_bundle_definitions = $this->templateDefinition->getDynamicBundleDefinitions();
+
+    $field_config = $this->createConfig($entity_type, $bundle, $dynamic_bundle_definitions, []);
+    $config_parts = explode('.', $this->templateDefinition->getKey('source'));
+
+    // Create ID for config.
+    array_pop($config_parts);
+    $config_id = "{$this->collection->getId()}_{$entity_type}_{$bundle}_{$field_name}";
+    $field_config['id'] = $config_id;
+    array_push($config_parts, $config_id);
```

Comment:
nit: $config_parts[] = $config_id is faster ;-)

Reply:
config_parts was removed